### ISSUE TITLE
Popup window hard to close without exectuting command

### ISSUE THREAD

**Issue by alerque** [liuchengxu/vim-clap#387]
**Environment (please complete the following information):**
 - OS: Neovim
 - (Neo)Vim version: latest stable
 - vim-clap version: latest master (has been true for months now)
 - Have you reproduced with a minimal vimrc: no
 - Have you updated to the latest plugin version: yes (isn't this redundant question)
 - Have you upgraded to/compiled the latest Rust binary: yes

**Describe the bug**
Over the last several months I keep finding myself with the Clap popup window and/or artifacts from in up over my buffer that are not part of my active work.

The issue seems to be that <kbd>Esc</kbd> does not effectively close the window, although it does usually change the input focus to something else. This creates all sorts of messes when I start trying to use Clap for something and then change my mind rather than opening a file or completing at action.

I have in fact started to bake <kbd>CTRL-W</kbd> <kbd>O</kbd> into muscle memory to make my buffer pane the only thing open as a way to clear artifacts from Clap leftover from failed attempts to escape from it.

I think the fix is to be more aggressive about cleanup up the UI when input focus changes, even if this wasn't the result of a completed action.

**Comment by liuchengxu** [comment]
Honestly speaking, I don't quite understand what your problem is. Are you talking about this issue #287?

**Comment by alerque** [comment]
Here's an example I _just_ ran into. I opened `:Clap files` to open something, but it didn't come up in the search (I was looking for something in the wrong repository). I realized my mistake and hit Escape. I ended up in a limbo mode where the main vim buffer has focus (note how my input cursor is down on the main vim cmd bar) and I can do things in that buffer, but artifacts from the Clap popup(s) are still there.

![image](https://user-images.githubusercontent.com/173595/79050614-d790f880-7c33-11ea-9b54-f2914acb5981.png)
